### PR TITLE
feat(transfer of vehicle ownership): Update data provider

### PR DESCRIPTION
**Pull request by johannaagma**
## What

Update data providers for Transfer of vehicle ownership

## Why

Make changes to data providers so the application works

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Comment by obmagnusson** [comment]
Could these providers be shared?

File: libs/application/template-api-modules/src/lib/modules/shared/api/vehicles/vehicles.service.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    if (!result || !result.length) {
+      if (showOwned && showCoOwned && !showOperated) {
+        throw new TemplateApiError(
+          {
+            title: coreErrorMessages.vehiclesEmptyListOwner,
+            summary: coreErrorMessages.vehiclesEmptyListOwner,
+          },
+          400,
+        )
+      } else if (showOwned && showCoOwned && !showOperated) {
```
Comment (obmagnusson) [inline libs/application/template-api-modules/src/lib/modules/shared/api/vehicles/vehicles.service.ts:45]: Fix lint error. Also not really clear how this error flow with the parameters is suppposed to work. You might want to consider some basic documentation on usage.

**Review by obmagnusson — CHANGES_REQUESTED** [review]

**Review by sigruntg — APPROVED** [review]

**Review by obmagnusson — APPROVED** [review]
Awesome! Another shared provider 🚀 💯

**Review by thordurhhh — APPROVED** [review]
💯

**Pull request merged**